app/components/e-checkbox.vue

```diff
@@ -0,0 +1,282 @@
+      <slot></slot>
+    </label>
+
+  </div>
+
+</template>
+
+<script>
+  import formStates from '@/mixins/form-states';
+
+  export default {
+    name: 'e-checkbox',
+    mixins: [formStates],
+    inheritAttrs: false,
+
+    model: {
+      /**
+       * Changes v-model behavior and use 'checked' instead of 'value' as prop.
+       * Avoids conflict with default value attribute.
+       */
+      prop: 'checked',
+      event: 'change',
+    },
+
+    props: {
+      /**
+       * Adds name attribute
+       * Note: is also used as id and for label (won't work without them)
+       */
+      name: {
+        type: String,
+        required: true,
+      },
+
+      /**
+       * Adds value attribute
+       */
+      value: {
+        type: String,
+        required: true,
+      },
+    },
+
+    // data() {
+    //   return {};
+    // },
+
+    computed: {
+      /**
+       * Sets value of component model to parent model
+       *
+       * @returns  {Boolean}   Status of the checkbox
+       */
+      internalValue: {
+        get() {
+          return this.checked;
+        },
+        set(value) {
+          this.$emit('change', value);
+        }
+      },
+
+      /**
+       * Defines state modifier classes
+       *
+       * @returns  {Object}   BEM classes
+       */
+      modifiers() {
+        return {
+          ...this.stateModifiers
+        };
+      }
+    },
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    methods: {
+      /**
+       * Emits state to parent and wrapper component
+       * Update "isChecked" state
+       *
+       * @param   {Boolean}  event   Field input
+       */
+      onChange(event) {
+        this.isChecked = event.target.checked;
+        this.$parent.$emit('change');
+      },
+
+      /**
+       * Emits focus to parent and wrapper component
+       * Update "hasFocus" state
+       */
+      onFocus() {
+        this.hasFocus = true;
+
+        this.$emit('focus');
+        this.$parent.$emit('focus');
+      },
+
+      /**
+       * Emits blur to parent and wrapper component
+       * Update "hasFocus" state
+       */
+      onBlur() {
+        this.hasFocus = false;
+
+        this.$emit('blur');
+        this.$parent.$emit('blur');
+      }
+    }
+    // render() {},
+  };
+</script>
+
+<style lang="scss">
+  .e-checkbox {
+    &__field {
+      // label styling
+      &:not(:checked),
+      &:checked {
+        position: absolute;
+        left: -9999px;
+        cursor: pointer;
+        -webkit-appearance: none;
+      }
+
+      &:not(:checked) + label,
+      &:checked + label {
+        position: relative;
+        padding-left: $spacing--25;
+        font-size: $font-size--14;
+        line-height: $font-size--18;
```

Comment:
you can define line-height in the font mixin, but I would use there a pixel value instead a $font-size variable